src/BuildScriptGeneratorCli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,104 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Common.Extensions;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Execute an arbitrary command in the default shell, in an environment " +
+        "with the best-matching platform tool versions.")]
+    internal class ExecCommand : CommandBase
```

Comment:
Tests? There are fair number of cases we need to test here. commands with quotes, multiple commands separated by `&&`, a script file itself

Reply:
working on them still; didn't want to block the review of this PR in the meantime

Reply:
I see you added only 1 E2E test. Can you add more scenarios as mentioned above?